runbotics/runbotics-scheduler/src/mail/mail.service.ts

```diff
@@ -53,18 +60,62 @@ export class MailService {
     }
 
     public async sendBotDisconnectionNotificationMail(bot: IBot, installationId: string) {
-        await this.sendMail({
-            to: bot.user.email ?? '',
-            subject: NOTIFICATION_MAIL_SUBJECT,
-            content: `Hello,\n\nBot 🤖 (${installationId}) has been disconnected.\n\nBest regards,\nRunBotics`,
-        });
+        const disconnectedBot = await this.botService.findById(bot.id);
+
+        if (disconnectedBot.subscribers && disconnectedBot.subscribers.length) {
+            const subscribers = disconnectedBot.subscribers;
+            const emailAddresses = subscribers.map(({ email }) => email).join(',');
+
+            await this.sendMail({
+                to: emailAddresses,
+                subject: NOTIFICATION_MAIL_SUBJECT,
+                content: `Hello,\n\nBot 🤖 (${installationId}) has been disconnected.\n\nBest regards,\nRunBotics`,
+            });
+        }
     }
 
     public async sendProcessFailureNotificationMail(process: IProcess, processInstance: IProcessInstance) {
-        await this.sendMail({
-            to: process.createdBy.email ?? '',
+        const failedProcess = await this.processService.findById(process.id);
+        const processCreatorEmail = failedProcess.createdBy.email;
+        const subscribers = failedProcess.subscribers;
+
+        const sendMailInput: SendMailInput = {
+            to: '',
             subject: NOTIFICATION_MAIL_SUBJECT,
             content: `Hello,\n\nProcess ⚙️ ${process.name} (${process.id}) has failed with status (${processInstance.status}).\n\nBest regards,\nRunBotics`,
-        });
+        };
+
+        if (!failedProcess.isPublic) {
+            const admins = await this.userService.findAllByRole(Role.ROLE_ADMIN);
+            const adminsIds = admins.map(admin => admin.id);
+            const filteredSubscribers =
+                subscribers && subscribers.length
+                    ? subscribers.reduce((acc, subscriber) => {
+                            if (adminsIds.includes(subscriber.id)) {
+                                acc.push(subscriber.email);
+                            }
+                            return acc;
+                        }, [])
+                    : [];
+
+            const emailAddresses = mergeArrays([
+                processCreatorEmail,
+                ...filteredSubscribers,
+            ]).join(',');
+
+            await this.sendMail({ ...sendMailInput, to: emailAddresses });
+        } else {
+            const subscribersAddresses =
+                subscribers && subscribers.length
+                    ? subscribers.map(({ email }) => email)
+                    : [];
+
+            const emailAddresses = mergeArrays([
+                processCreatorEmail,
+                ...subscribersAddresses,
+            ]).join(',');
+
+            await this.sendMail({...sendMailInput, to: emailAddresses });
+        }
```

Comment:
I would move the logic to organize subscribers to a separate function so you don't have huge if statement here and logic can be shared